src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -203,12 +204,23 @@ public void register(VillagerProfession prof)
     }
     private void register(VillagerProfession prof, int id)
     {
-        professions.register(id, prof.name, prof);
+        prof.setId(id);
+        if(professionsByName.containsKey(prof.getName()))
+        {
+            throw new IllegalArgumentException("Cannot Register the same Profession Name twice");
+        }
+        if(FMLCommonHandler.instance().getEffectiveSide() == Side.CLIENT)
+        {
+            registerVillagerSkin(prof.id, prof.texture);
+        }
+        professionsById.put(prof.id, prof);
+        professionsByName.put(prof.getName(), prof);
     }
 
     private boolean hasInit = false;
-    private FMLControlledNamespacedRegistry<VillagerProfession> professions = GameData.createRegistry("villagerprofessions", VillagerProfession.class, 0, 1024);
-
+//    private FMLControlledNamespacedRegistry<VillagerProfession> professions = GameData.createRegistry("villagerprofessions", VillagerProfession.class, 0, 1024);
+    private HashMap<Integer, VillagerProfession> professionsById = new HashMap<Integer, VillagerProfession>();
+    private HashMap<String, VillagerProfession> professionsByName = new HashMap<String, VillagerProfession>();
```

Comment:
No, just no, FMLControlledNamespacedRegistry is exactly this. A {Object, String, Int} -> Object